### ISSUE TITLE
bug report: 

### ISSUE THREAD

**Issue by Minikea** [docker-mailserver/docker-mailserver#3309]
### Preliminary Checks

- [X] I checked that all ports are open and not blocked by my ISP / hosting provider.
- [X] I know that SSL errors are likely the result of a wrong setup on the user side and not caused by DMS itself. I'm confident my setup is correct.
- [X] I searched the issue tracker but was unable to find my issue.
- [X] I read the [extended documentation in general](https://docker-mailserver.github.io/docker-mailserver/latest/) but found nothing to resolve the issue.
- [X] I read the [documentation on debugging](https://docker-mailserver.github.io/docker-mailserver/latest/config/debugging/), tried the proposed steps to debug the problem, but was still unable to resolve the issue.
- [X] I have read this project's [Code of Conduct](https://github.com/docker-mailserver/docker-mailserver/blob/master/CODE_OF_CONDUCT.md) and I agree

### Affected Component(s)

no mail received

### What happened and when does this occur?

no mail received since a month. the mailbox is not a critical one and is not consulted very often.

```May  5 08:57:46 mail postfix/qmgr[664]: warning: connect to transport private/`lmtp: No such file or directory
May  5 08:57:46 mail postfix/qmgr[664]: 658596CCE5: removed
May  5 08:57:46 mail postfix/error[2071]: 4668B6CCEF: to=<xxx@domain.com>, relay=none, delay=0.02, delays=0/0.01/0/0.01, dsn=4.3.0, status=deferred (mail transport unavailable)```
in log and
``` virtual_transport = `lmtp:unix:private/dovecot-lmtp` ``` in resulting config file
is the hyphen not correctly interpreted?

### What did you expect to happen?

mail not deferred infinitely

### How do we replicate the issue?

1.
2.
3.
...

### DMS version

docker-mailserver 12.0.0

### What operating system is DMS running on?

Linux

### Which operating system version?

Debian 11

### What instruction set architecture is DMS running on?

AMD64 / x86_64

### What container orchestration tool are you using?

Docker

### docker-compose.yml

```yml
version: '2'
services:
  mail:
    image: ghcr.io/docker-mailserver/docker-mailserver:latest
    hostname: ${HOSTNAME}
    domainname: ${DOMAINNAME}
    container_name: ${CONTAINER_NAME}
    ports:
        - "25:25"
        - "143:143"
        - "465:465"
        - "587:587"
        - "993:993"
    volumes:
        - ./maildata:/var/mail
        - mailstate:/var/mail-state
        - maillogs:/var/log/mail
        - ./config/:/tmp/docker-mailserver/
        - /etc/letsencrypt:/etc/letsencrypt
        - /etc/localtime:/etc/localtime:ro
    labels:
        - "com.centurylinklabs.watchtower.enable=true"
    env_file:
        - .env
        - env-mailserver
    cap_add:
        - NET_ADMIN
        - SYS_PTRACE
    restart: always
    dns:
        - 151.80.222.79
        - 51.15.98.97
        - 9.9.9.9
```

### Relevant log output

```Text
mail               | May  5 08:57:34 mail postfix/postscreen[2040]: CONNECT from [209.85.160.44]:41257 to [172.18.0.2]:25
mail               | May  5 08:57:40 mail postfix/postscreen[2040]: PASS NEW [209.85.160.44]:41257
mail               | May  5 08:57:40 mail postfix/smtpd[2050]: connect from mail-oa1-f44.google.com[209.85.160.44]
mail               | May  5 08:57:40 mail postfix/smtpd[2050]: Anonymous TLS connection established from mail-oa1-f44.google.com[209.85.160.44]: TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits) key-exchange X25519 server-signature RSA-PSS (2048 bits) server-digest SHA256
mail               | May  5 08:57:41 mail policyd-spf[2058]: prepend Received-SPF: Pass (mailfrom) identity=mailfrom; client-ip=209.85.160.44; helo=mail-oa1-f44.google.com; envelope-from=test@gmail.com; receiver=<UNKNOWN>
mail               | May  5 08:57:41 mail postfix/smtpd[2050]: 658596CCE5: client=mail-oa1-f44.google.com[209.85.160.44]
mail               | May  5 08:57:41 mail postfix/cleanup[2059]: 658596CCE5: message-id=<CAMJW86+pWLKNGyfz51Gs=SPEW6xXxX+k=G7Jd6QiY-oSVuNX+g@mail.gmail.com>
mail               | May  5 08:57:41 mail opendkim[505]: 658596CCE5: mail-oa1-f44.google.com [209.85.160.44] not internal
mail               | May  5 08:57:41 mail opendkim[505]: 658596CCE5: not authenticated
mail               | May  5 08:57:41 mail opendkim[505]: 658596CCE5: DKIM verification successful
mail               | May  5 08:57:41 mail opendkim[505]: 658596CCE5: s=20221208 d=gmail.com a=rsa-sha256 SSL
mail               | May  5 08:57:41 mail opendmarc[511]: 658596CCE5: gmail.com pass
mail               | May  5 08:57:41 mail postfix/qmgr[664]: 658596CCE5: from=<test@gmail.com>, size=3346, nrcpt=1 (queue active)
mail               | May  5 08:57:46 mail postfix/smtpd-amavis/smtpd[2070]: connect from localhost[127.0.0.1]
mail               | May  5 08:57:46 mail postfix/smtpd-amavis/smtpd[2070]: 4668B6CCEF: client=localhost[127.0.0.1]
mail               | May  5 08:57:46 mail postfix/cleanup[2059]: 4668B6CCEF: message-id=<CAMJW86+pWLKNGyfz51Gs=SPEW6xXxX+k=G7Jd6QiY-oSVuNX+g@mail.gmail.com>
mail               | May  5 08:57:46 mail postfix/qmgr[664]: 4668B6CCEF: from=<test@gmail.com>, size=3815, nrcpt=1 (queue active)
mail               | May  5 08:57:46 mail postfix/qmgr[664]: warning: connect to transport private/`lmtp: No such file or directory
mail               | May  5 08:57:46 mail amavis[683]: (00683-01) Passed CLEAN {RelayedInbound}, [209.85.160.44]:41257 [209.85.160.44] <test@gmail.com> -> <user@domain.com>, Queue-ID: 658596CCE5, Message-ID: <CAMJW86+pWLKNGyfz51Gs=SPEW6xXxX+k=G7Jd6QiY-oSVuNX+g@mail.gmail.com>, mail_id: dR0vwWxA9mXc, Hits: -0.198, size: 3604, queued_as: 4668B6CCEF, 4641 ms
mail               | May  5 08:57:46 mail postfix/smtp-amavis/smtp[2061]: 658596CCE5: to=<user@domain.com>, relay=127.0.0.1[127.0.0.1]:10024, delay=5.3, delays=0.62/0.01/0.01/4.6, dsn=2.0.0, status=sent (250 2.0.0 from MTA(smtp:[127.0.0.1]:10025): 250 2.0.0 Ok: queued as 4668B6CCEF)
mail               | May  5 08:57:46 mail postfix/qmgr[664]: 658596CCE5: removed
mail               | May  5 08:57:46 mail postfix/error[2071]: 4668B6CCEF: to=<user@domain.com>, relay=none, delay=0.02, delays=0/0.01/0/0.01, dsn=4.3.0, status=deferred (mail transport unavailable)
mail               | May  5 08:58:14 mail postfix/smtpd[2050]: disconnect from mail-oa1-f44.google.com[209.85.160.44] ehlo=2 starttls=1 mail=1 rcpt=1 bdat=1 quit=1 commands=7
```

### What level of experience do you have with Docker and mail servers?

- [ ] I am inexperienced with docker
- [X] I am rather experienced with docker
- [ ] I am inexperienced with mail servers
- [ ] I am rather experienced with mail servers
- [ ] I am uncomfortable with the CLI
- [ ] I am rather comfortable with the CLI

**Comment by Minikea** [comment]
if I change the env file to bypass the default configuration and remove the `
I have this error : 
`mail               | May  5 09:23:14 mail postfix/lmtp[731]: 2862D6CCEE: to=<user@domain.net>, relay=none, delay=0.02, delays=0/0.01/0/0, dsn=4.4.1, status=deferred (connect to mail.domain.net[private/dovecot-lmtp]: No such file or directory)`

**Comment by Minikea** [comment]
I think I got this one:
there is a change in env file:
before:
```# This Option is activating the Usage of POSTFIX_DAGENT to specify a lmtp client different from default dovecot socket.
# empty => disabled
# 1 => enabled
ENABLE_POSTFIX_VIRTUAL_TRANSPORT=

# Enabled by ENABLE_POSTFIX_VIRTUAL_TRANSPORT. Specify the final delivery of postfix
#
# empty => fail
# `lmtp:unix:private/dovecot-lmtp` (use socket)
# `lmtps:inet:<host>:<port>` (secure lmtp with starttls, take a look at https://sys4.de/en/blog/2014/11/17/sicheres-lmtp-mit-starttls-in-dovecot/)
# `lmtp:<kopano-host>:2003` (use kopano as mailstore)
# etc.
POSTFIX_DAGENT=`lmtp:unix:private/dovecot-lmtp` ```
and the new one :
```# Configure Postfix `virtual_transport` to deliver mail to a different LMTP client (default is a dovecot socket).
# Provide any valid URI. Examples:
#
# empty => `lmtp:unix:/var/run/dovecot/lmtp` (default, configured in Postfix main.cf)
# `lmtp:unix:private/dovecot-lmtp` (use socket)
# `lmtps:inet:<host>:<port>` (secure lmtp with starttls)
# `lmtp:<kopano-host>:2003` (use kopano as mailstore)
POSTFIX_DAGENT=```
the socket name change and the env variable is not activated the same way so it activated the old one with the wrong name.

**Comment by polarathene** [comment]
The socket rename was mentioned in the [release notes for v12](https://github.com/docker-mailserver/docker-mailserver/releases/tag/v12.0.0), but may not have been as obvious sorry.

The removal of `ENABLE_POSTFIX_VIRTUAL_TRANSPORT` was not mentioned though (sorry), the PR related to that change is [detailed here](https://github.com/docker-mailserver/docker-mailserver/pull/3004).

> `POSTFIX_DAGENT=`lmtp:unix:private/dovecot-lmtp` 

This was not assigned in the ENV file, maybe you added it yourself at some point as it was never assigned a value in this file according to `git blame`. The comments just show possible config values you could use. While the `empty` one added is not a change and has always been the case as our default.

`POSTFIX_DAGENT` is for changing this value, mostly when you're using an alternative than Dovecot (_or at least not the Dovecot we provide within our image_).

---

Be mindful of updates, your compose config looks to be falling out of sync with what we ship (_shouldn't cause any major issues though_), and you're not version pinning, `:latest` tag puts you at risk of breaking change updates. Although it seems that isn't too problematic for you at least :sweat_smile:
